library/Director/Web/Form/Element/Dictionary.php

```diff
@@ -79,7 +74,7 @@ public function setValue($value)
         if ($value instanceof \stdClass) {
             $value = json_decode(json_encode($value), true);
         }
-        return parent::setValue($value);
+        return parent::setValue($this->applyOnStructure($value));
```

Comment:
The `setValue` is called _after_ the validation, we tried to have the "application thing" in one place but the way Icinga is built doesn't allows this.